cognite/client/config.py

```diff
@@ -47,41 +57,39 @@ def __init__(self) -> None:
         self.max_workers: int = 5
         self.silence_feature_preview_warnings: bool = False
 
-    @classmethod
-    def load(cls, config: dict | str) -> GlobalConfig:
-        """Loads a dictionary of configuration fields into a client config object.
+    def apply_settings(self, settings: dict[str, Any] | str) -> None:
+        """Apply settings to the global configuration object from a YAML/JSON string or dict.
 
-        Note: This must be done before instantiating a CogniteClient for the configuration to take effect.
+        Warning:
+            This must be done before instantiating a CogniteClient for the configuration to take effect.
 
         Args:
-            config (dict | str): A dictionary or dictionary parsable string containing configuration values defined in the GlobalConfig class.
-
-        Returns:
-            GlobalConfig: A global configuration object.
+            settings (dict[str, Any] | str): A dictionary or YAML/JSON string containing configuration values defined in the GlobalConfig class.
 
         Examples:
-                Create a global config object from a dictionary input:
 
-                >>> from cognite.client.config import GlobalConfig
-                >>> config = {
+            Create a global config object from a dictionary input:
+
+                >>> from cognite.client import global_config
+                >>> settings = {
                 ...     "max_retries": 5,
                 ...     "disable_ssl": True,
                 ... }
-                >>> global_config = GlobalConfig.load(config)
+                >>> global_config.apply_settings(settings)
         """
 
-        loaded = load_dict_or_str(config)
+        loaded = load_resource_to_dict(settings)
 
-        global_config = cls()
         for key, value in loaded.items():
             if not hasattr(global_config, key):
                 raise ValueError(f"Invalid key in global config: {key}")
             if key == "default_client_config":
-                global_config.default_client_config = ClientConfig.load(value)
+                if isinstance(value, ClientConfig):
+                    self.default_client_config = value
+                else:
+                    self.default_client_config = ClientConfig.load(value)
```

Comment:
I'm glad I had the epiphany because my brain was thinking so much in just a dict[str, str] mode (ie the loading from yaml scenario) 🤣 I'll add a test for it too